Make OC_SubAdmin non static

rullzer opened the pull request:
Replaces #17773 for proper CI tests

This feature is present currently in core but not tested (or mockable) at all. If it is there we should properly test it.

Remaining point is if we should just add this to the groupManager (but not to the public interface)? Since it had to be somewhere I think this is the most logical choice. But suggestions are welcome.

CC: @PVince81 @MorrisJobke @DeepDiver1975 @LukasReschke @nickvergessen

rullzer committed:
New \OC\SubAdmin class

* DI
* Tests
* moved OC_SubAdmin to legacy
* Added to private OC\GroupManager

PVince81 commented:
Would it be possible to rewire the static class to call the non-static one to avoid duplicating logic ?

rullzer commented:
Yes that should be possible... let see if a simple commit can make that happen :)

rullzer committed:
Legacy OC_SubAdmin is now just a wrapper

rullzer committed:
Fix unit test

Now that OC_SubAdmin is just a wrapper around OC\SubAdmin some unit
tests had to be fixed because they expected different behaviour.

Eventually they should move to properly mocked instances of OC\SubAdmin
of course

rullzer commented:
@PVince81 OC_SubAdmin is now just a wrapper around OC\SubAdmin.. failing unit tests because of that have been addressed.

rullzer commented:
OCI fails on the background jobs again...

nickvergessen commented:
Happens from time to time, just ignore or rebase

MorrisJobke commented:
I did some basic testing, everything still works :+1:

PVince81 commented:
:+1: still works

DeepDiver1975 commented:
needs rebase ...
